client/state/site-settings/actions.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	SITE_SETTINGS_RECEIVE,
+	SITE_SETTINGS_REQUEST,
+	SITE_SETTINGS_REQUEST_FAILURE,
+	SITE_SETTINGS_REQUEST_SUCCESS,
+	SITE_SETTINGS_SAVE,
+	SITE_SETTINGS_SAVE_FAILURE,
+	SITE_SETTINGS_SAVE_SUCCESS,
+	SITE_SETTINGS_UPDATE
+} from 'state/action-types';
+
+/**
+ * Returns an action object to be used in signalling that site settings have been received.
+ *
+ * @param  {Number} siteId Site ID
+ * @param  {Object} settings The site settings object
+ * @return {Object}        Action object
+ */
+export function receiveSiteSettings( siteId, settings ) {
+	return {
+		type: SITE_SETTINGS_RECEIVE,
+		siteId,
+		settings
+	};
+}
+
+/**
+ * Returns an action object to be used in signalling that some site settings have been update.
+ *
+ * @param  {Number} siteId Site ID
+ * @param  {Object} settings The updated site settings
+ * @return {Object}        Action object
+ */
+export function updateSiteSettings( siteId, settings ) {
+	return {
+		type: SITE_SETTINGS_UPDATE,
+		siteId,
+		settings
+	};
+}
+
+/**
+ * Returns an action thunk which, when invoked, triggers a network request to
+ * retrieve site settings
+ *
+ * @param  {Number} siteId Site ID
+ * @return {Function}      Action thunk
+ */
+export function requestSiteSettings( siteId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SITE_SETTINGS_REQUEST,
+			siteId
+		} );
+
+		return new Promise( ( resolve, reject ) => {
+			wpcom.undocumented().settings( siteId, ( error, data ) => {
+				error ? reject( error ) : resolve( data );
+			} );
+		} ).then( ( { name, description, settings } ) => {
+			const savedSettings = {
+				...settings,
+				blogname: name,
```

Comment:
I don't think we should ignore these properties (at least at the current state of the endpoint), because to update the name and description, we 'POST' to this same endpoint blogname and blogdescription. So I'm doing this normalization to make `saveSettings` and `requestSettings` consistent: (We can post the settings we retrieve directly or by changing the attributes values only and not the object shape).